source/Octopus.Manager.Tentacle/TentacleConfiguration/SetupWizard/TentacleSetupWizardModel.cs

```diff
@@ -676,16 +679,16 @@ Task<IOctopusAsyncClient> CreateClient()
             return OctopusAsyncClient.Create(endpoint, new OctopusClientOptions());
         }
 
-        async Task LoadSpaceSpecificData(IOctopusAsyncClient client)
+        async Task LoadSpaceSpecificData(IOctopusAsyncClient client, CancellationToken cancellationToken)
         {
             var spaceRepository = await new SpaceRepositoryFactory().CreateSpaceRepository(client, SelectedSpace);
             await LoadDataFromSpace(_ =>
             {
                 /*users aren't actually interested in these progress messages, and we have nowhere to display them*/
-            }, spaceRepository);
+            }, spaceRepository, cancellationToken);
         }
 
-        async Task LoadSpaceData(Func<IOctopusAsyncClient, Task> loadAction)
+        async Task LoadSpaceData(Func<IOctopusAsyncClient, CancellationToken, Task> loadAction, CancellationToken cancellationToken)
```

Comment:
@APErebus if you look at this method, you'll see how it's handling exceptions.

Related to this thread: https://github.com/OctopusDeploy/OctopusTentacle/pull/622#discussion_r1348082515